### PR TITLE
Adding Linux performance testing CI leg

### PR DESCRIPTION
**Pull request by ghost**
**build/build.sh** 
case-sensitive changes; switch fixes

**build/uploadperftobenchview.cmd** 
clean up some logic

**build/uploadperftobenchview.sh** 
Linux performance results uploader script

**perf.groovy** 
add Linux Performance  CI run

Addresses: https://github.com/dotnet/sdk/issues/2150
Replaces: https://github.com/dotnet/sdk/pull/2300

@DrewScoggins 

Successful Linux Performance run:
https://ci2.dot.net/job/dotnet_sdk/job/perf/job/release_2.1.3xx/job/GenPRTest/job/SDK_Perf_Ubuntu_16.04_x64_prtest/28/console

Successful Win-x86 Performance run:
https://ci2.dot.net/job/dotnet_sdk/job/perf/job/release_2.1.3xx/job/GenPRTest/job/SDK_Perf_Windows_NT_x86_prtest/2/

Successful Win-x64 Performance run:
https://ci2.dot.net/job/dotnet_sdk/job/perf/job/release_2.1.3xx/job/GenPRTest/job/SDK_Perf_Windows_NT_x64_prtest/2/

### CODE REVIEW

**Comment by johnbeisner** [comment]
@dotnet-bot test CI please

**Comment by johnbeisner** [comment]
@dotnet-bot test CI please

File: perf.groovy
Comment (mmitche) [inline perf.groovy]: "\${WORKSPACE}/artifacts/${config}/TestResults/Performance"

File: build/uploadperftobenchview.sh
Comment (dsplaisted) [inline build/uploadperftobenchview.sh]: This seems like the wrong way to check the values of environment variables.  I think the right way is something like this:

    if [ "$runType" = private ]; then

File: build/uploadperftobenchview.sh
Comment (dsplaisted) [inline build/uploadperftobenchview.sh]: I think a better way to do this may be something like this:

    if [[ -z $perfWorkingDirectory ]]; then

File: build/uploadperftobenchview.sh
Comment (tannergooding) [inline build/uploadperftobenchview.sh]: Please run this through shellcheck (there is an online form via `https://www.shellcheck.net`)

This is reporting a number of items that should be fixed before merging:
```
$ shellcheck myscript
 
Line 53:
rm -r -f $perfWorkingDirectory/Microsoft.BenchView.JSONFormat > /dev/null 2>&1
         ^-- SC2154: perfWorkingDirectory is referenced but not assigned.
         ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 55:
if [ ! -d $perfWorkingDirectory/Microsoft.BenchView.JSONFormat ]; then
          ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 58:
    curl -o $perfWorkingDirectory/benchview.zip http://benchviewtestfeed.azurewebsites.net/api/v2/package/microsoft.benchview.jsonformat/0.1.0-pre024
            ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 59:
    unzip -q -o $perfWorkingDirectory/benchview.zip -d $perfWorkingDirectory/Microsoft.BenchView.JSONFormat
                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 71:
benchViewName="SDK perf $OS $architecture $configuration $runType $GIT_BRANCH_WITHOUT_ORIGIN"
                            ^-- SC2154: architecture is referenced but not assigned.
                                          ^-- SC2154: configuration is referenced but not assigned.
                                                         ^-- SC2154: runType is referenced but not assigned.
 
Line 74:
    benchViewName="$benchViewName $BenchviewCommitName"
                                  ^-- SC2154: BenchviewCommitName is referenced but not assigned.
 
Line 80:
echo BenchViewName: $benchViewName
                    ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 82:
echo Creating: $perfWorkingDirectory/submission-metadata.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 84:
                    -o "$perfWorkingDirectory/submission-metadata.json" || { echo Failed to create: $perfWorkingDirectory/submission-metadata.json && exit 1 ; }
>>                                                                                                  ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 86:
echo Creating: $perfWorkingDirectory/build.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 87:
python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/build.py" git --branch $GIT_BRANCH_WITHOUT_ORIGIN --type "$runType" \
>>                                                                                           ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 88:
                   -o "$perfWorkingDirectory/build.json" || { echo Failed to create: $perfWorkingDirectory/build.json && exit 1 ; }
>>                                                                                   ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 90:
echo Creating: $perfWorkingDirectory/machinedata.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 92:
                   -o "$perfWorkingDirectory/machinedata.json" || { echo Failed to create: $perfWorkingDirectory/machinedata.json && exit 1 ; }
>>                                                                                         ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 94:
echo Creating: $perfWorkingDirectory/measurement.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 95:
find $perfWorkingDirectory -maxdepth 1 -type f -name "*.xml" -exec echo Processing: "{}" \;
     ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 96:
find $perfWorkingDirectory -maxdepth 1 -type f -name "*.xml" -exec python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/measurement.py" xunitscenario "{}" --better desc --append \
     ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 99:
echo Creating: $perfWorkingDirectory/submission.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 112:
echo Uploading: $perfWorkingDirectory/submission.json
                ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 113:
python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/upload.py" "$perfWorkingDirectory/submission.json" --container coreclr || { echo Failed to upload: $perfWorkingDirectory/submission.json && exit 1 ; }
>>                                                                                                                                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.

$ 
```

File: build/uploadperftobenchview.sh
Comment (DrewScoggins) [inline build/uploadperftobenchview.sh]: Where are the result XML files being dropped and where are we running this command from? I assume the command is executed from the workspace root, but want to make sure. I think it might be good here to do some kind of error checking that we are still finding the results and making sure that we are generating the measurement.json with all of the data.

File: build/uploadperftobenchview.sh
Comment (johnbeisner) [reply build/uploadperftobenchview.sh]: @dsplaisted 
We are checking to see if the value is set - I can add the check for existence.

File: build/uploadperftobenchview.sh
Comment (johnbeisner) [reply build/uploadperftobenchview.sh]: "making sure that we are generating the measurement.json" - we do a check at each call to "measurement.py" Is that not sufficient?

File: build/uploadperftobenchview.sh
Comment (johnbeisner) [reply build/uploadperftobenchview.sh]: done...

File: build/uploadperftobenchview.sh
Comment (johnbeisner) [reply build/uploadperftobenchview.sh]: done...

File: build/uploadperftobenchview.sh
Comment (johnbeisner) [reply build/uploadperftobenchview.sh]: done...

File: perf.groovy
Comment (johnbeisner) [reply perf.groovy]: done...

**Comment by johnbeisner** [comment]
@dotnet-bot test CI please

**Comment by johnbeisner** [comment]
@dotnet-bot test CI please

**Comment by johnbeisner** [comment]
@dotnet/dotnet-cli 
Can I get an approval on this PR.

File: build/uploadperftobenchview.sh
Comment (DrewScoggins) [reply build/uploadperftobenchview.sh]: It's more I am worried that we could change the way that the results are produced and now the find command does not return all of the results to be processed. Given this is likely a small likelihood, but just something that I noticed.

**Comment by DrewScoggins** [comment]
LGTM

**Review by dsplaisted — APPROVED** [review]
Approving based on @DrewScoggins' sign-off

**Pull request merged**
